pf9/modules/analytics_utils.py

```diff
@@ -25,6 +25,15 @@ def load_segment_session(self, segment_session, segment_event_properties, segmen
         self.ctx.params["segment_subcommand"] = segment_subcommand
         self.ctx.params["segment_step_count"] = 1
 
+    def reload_segment_session_with_auth(self):
+        """
+            Reload Segment Session with DU URL and Keystone User ID after keystone Auth
+        """
```

Comment:
Could you expand more why this is needed? Here or in the PR description.

Reply:
This is needed to add keystone user_id and account_url to the segment event properties post initialization. As, we want to capture pre Auth Failures/Errors, we should be able to send Events to segment pre auth and pre config load, during which we wont have keystone user_id / du_account_url.